x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -44,6 +45,13 @@ func init() {
 // interface methods except for Fetch.
 type MetricSet struct {
 	*aws.MetricSet
+	Tags []Tag `config:"tags_filter"`
+}
+
+// Tag holds a configuration specific for ec2 metricset.
+type Tag struct {
+	Name  string `config:"name"`
+	Value string `config:"value"`
```

Comment:
I guess that both fields are required, consider adding `required` to both.

Reply:
Yeah I didn't add it because I'm debating if we should support other use case. For example, only given tag name, and support regex for both name and value. But for given both name and value, is the basic functionality here.

Reply:
Oh ok, this kind of though also answers the question that using an array of structs is better than a map for the config of the tags :slightly_smiling_face: